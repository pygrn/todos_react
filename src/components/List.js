import React from 'react'
import Row from './Row'
import apiclient from '../lib/apiclient'

class List extends React.Component {
  constructor(props) {
    super(props)
    this.state = {todos: [], loading: false}
  }

  componentDidMount() {
    this.updateList()
  }

  updateList = () => {
    this.setState({loading: true})
    apiclient.getTodos().then(todos => {
      if (todos) {
        this.setState({todos: todos, loading: false})
      }
    })
  }

  render () {
    return (
      <div>
        <h1>List</h1>
        <table className="table striped">
          <thead>
            <tr>
              <th>Description</th>
              <th>Due Date</th>
            </tr>
          </thead>
          <tbody>
            {this.state.todos.map(todo => <Row todo={todo} key={todo.pk}/>)}
          </tbody>
        </table>
        <br/><button className="btn waves-effect waves-light" onClick={this.updateList} disabled={this.state.loading}>
          {this.state.loading ? 'Loading' : 'Refresh'}
        </button>
      </div>
    )
  }
}

export default List