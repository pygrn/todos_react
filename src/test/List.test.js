import React from 'react'
import List from '../components/List'
import renderer from 'react-test-renderer'

jest.mock('../lib/apiclient')

test('List', () => {
    const component = renderer.create(<List/>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})