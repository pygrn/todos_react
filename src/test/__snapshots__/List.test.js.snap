// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List 1`] = `
<div>
  <h1>
    List
  </h1>
  <table
    className="table striped"
  >
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Due Date
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <br />
  <button
    className="btn waves-effect waves-light"
    disabled={true}
    onClick={[Function]}
  >
    Loading
  </button>
</div>
`;
